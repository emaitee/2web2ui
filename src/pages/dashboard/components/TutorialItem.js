import React from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

import { Icon } from '@sparkpost/matchbox';
import styles from './TutorialItem.module.scss';

const TutorialItem = ({
  label,
  children,
  labelLink,
  completed
}) => {
  const itemClasses = classnames(styles.Item, completed && styles.completed);

  const labelMarkup = (labelLink && !completed)
    ? <h6 className={classnames(styles.Label, styles.link)}><Link to={labelLink}>{ label } <Icon name='ArrowRight'/></Link></h6>
    : <h6 className={styles.Label}>{ label }</h6>;

  const completedMarkup = completed
    ? <Icon name='Check' size={24} className={styles.Completed} />
    : null;

  return (
    <div className={itemClasses}>
      { completedMarkup }
      <div className={styles.Content}>
        { labelMarkup }
        {!completed && <div className={styles.HelpText}>{ children }</div>}
      </div>
    </div>
  );
};

export default TutorialItem;
