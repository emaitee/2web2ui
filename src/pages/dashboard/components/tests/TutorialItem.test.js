import { shallow } from 'enzyme';
import React from 'react';

import TutorialItem from '../TutorialItem';

describe('Component: TutorialItem', () => {
  const props = {
    label: 'my item',
    completed: false,
    children: ['Evie', 'Levon']
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TutorialItem {...props} />);
  });

  it('should render correctly by default', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show checkmark when task is completed', () => {
    wrapper.setProps({ completed: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should ignore a label link and children when the item is completed', () => {
    wrapper.setProps({ labelLink: '/go/to/here', completed: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the link and children when not completed', () => {
    wrapper.setProps({ labelLink: '/go/to/here' });
    expect(wrapper).toMatchSnapshot();
  });

});
