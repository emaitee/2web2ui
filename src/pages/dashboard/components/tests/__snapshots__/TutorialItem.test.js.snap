// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TutorialItem should ignore a label link and children when the item is completed 1`] = `
<div
  className=""
>
  <Icon
    name="Check"
    size={24}
  />
  <div>
    <h6>
      my item
    </h6>
  </div>
</div>
`;

exports[`Component: TutorialItem should render correctly by default 1`] = `
<div
  className=""
>
  <div>
    <h6>
      my item
    </h6>
    <div>
      Evie
      Levon
    </div>
  </div>
</div>
`;

exports[`Component: TutorialItem should render the link and children when not completed 1`] = `
<div
  className=""
>
  <div>
    <h6
      className=""
    >
      <Link
        replace={false}
        to="/go/to/here"
      >
        my item
         
        <Icon
          name="ArrowRight"
          size={18}
        />
      </Link>
    </h6>
    <div>
      Evie
      Levon
    </div>
  </div>
</div>
`;

exports[`Component: TutorialItem should show checkmark when task is completed 1`] = `
<div
  className=""
>
  <Icon
    name="Check"
    size={24}
  />
  <div>
    <h6>
      my item
    </h6>
  </div>
</div>
`;
